src/test/java/selenium/patterns/LoginTestWithPageFactory.java

```diff
@@ -0,0 +1,41 @@
+package selenium.patterns;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import selenium.Driver;
+import selenium.TestUser;
+import selenium.pages.LoginPageWithPF;
+import selenium.pages.ProductPageWithPF;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class LoginTestWithPageFactory {
+    private static final String INVENTORY_URL = "https://www.saucedemo.com/inventory.html";
+    private LoginPageWithPF loginPage;
+    private ProductPageWithPF productPage;
+    private TestUser testUser;
+
+    @BeforeEach
+    public  void setup() {
+        loginPage = new LoginPageWithPF();
+        productPage = new ProductPageWithPF();
```

Comment:
no need to init these pages in the setup method

Reply:
@KBugrimCS Please provide more detailed explanations on why this pages shouldn't be initialized?

Reply:
Done